MDNetworking/Misc/MDReplicatorNetworkKeyIdMap.cs

```diff
@@ -13,8 +14,11 @@ class MDReplicatorNetworkKeyIdMap
 
         public void AddNetworkKeyIdPair(uint id, string key)
         {
-            NetworkIDToKeyMap.Add(id, key);
-            KeyToNetworkIdMap.Add(key, id);
+            if (NetworkIDToKeyMap.ContainsKey(id) == false)
```

Comment:
I'm not sure if it's expected that the id could be in the map but I was hitting an exception because it existed already (same for the buffer below)

Reply:
That is very weird, It should be impossible for the ID to already be in the map as far as I thought from the way the logic of this works. If the ID exists in the map we have a bigger problem though, just ignoring it with contains won't be good enough I think as this may cause replication to fail. If you are able to reproduce this maybe you can debug it and check if the key for the ID in the list is the same as the one being sent in, because if they are not we got a problem.

Reply:
Looks like it's the same:
```log
Tried to add key /root/GameSession/@@16#PlayerHealth for id 21 but it already has key /root/GameSession/@@16#PlayerHealth
```
it only happens on the client that joined